controllers/reconcilers/grafana/deployment_reconciler.go

```diff
@@ -211,7 +225,59 @@ func getContainers(cr *v1beta1.Grafana, scheme *runtime.Scheme, vars *v1beta1.Op
 	return containers
 }
 
-func getDeploymentSpec(cr *v1beta1.Grafana, deploymentName string, scheme *runtime.Scheme, vars *v1beta1.OperatorReconcileVars) v12.DeploymentSpec {
+// getGrafanaContainerSecurityContext provides default securityContext for grafana container
+func getGrafanaContainerSecurityContext(openshiftPlatform bool) *v1.SecurityContext {
+	capability := &v1.Capabilities{
+		Drop: []v1.Capability{
+			"ALL",
+		},
+	}
+	if openshiftPlatform {
+		return &v1.SecurityContext{
+			AllowPrivilegeEscalation: model.BoolPtr(false),
+			ReadOnlyRootFilesystem:   model.BoolPtr(true),
+			Privileged:               model.BoolPtr(false),
+			RunAsNonRoot:             model.BoolPtr(true),
+			Capabilities:             capability,
+		}
+	}
+	return &v1.SecurityContext{
+		AllowPrivilegeEscalation: model.BoolPtr(false),
+		ReadOnlyRootFilesystem:   model.BoolPtr(true),
+		Privileged:               model.BoolPtr(false),
+		RunAsNonRoot:             model.BoolPtr(true),
+		RunAsUser:                model.IntPtr(10001),
+		RunAsGroup:               model.IntPtr(10001),
+		Capabilities:             capability,
+	}
+}
+
+func getReadinessProbe(cr *v1beta1.Grafana) *v1.Probe {
+	return &v1.Probe{
+		ProbeHandler: v1.ProbeHandler{
+			HTTPGet: &v1.HTTPGetAction{
+				Path:   GrafanaHealthEndpoint,
+				Port:   intstr.FromInt(GetGrafanaPort(cr)),
+				Scheme: v1.URISchemeHTTP,
+			},
+		},
+		InitialDelaySeconds: ReadinessProbeInitialDelaySeconds,
+		TimeoutSeconds:      ReadinessProbeTimeoutSeconds,
+		PeriodSeconds:       ReadinessProbePeriodSeconds,
+		SuccessThreshold:    ReadinessProbeSuccessThreshold,
+		FailureThreshold:    ReadinessProbeFailureThreshold,
+	}
+}
+
+func getPodSecurityContext() *v1.PodSecurityContext {
+	return &v1.PodSecurityContext{
+		SeccompProfile: &v1.SeccompProfile{
+			Type: "RuntimeDefault",
```

Comment:
I haven't had any experience running pods with SeccompProfile enabled, so my concerns are based on reading the docs:
- The functionality was moved to beta only 1.25, which is a recent release, so most users won't have it enabled by default;
- (from docs) "To use seccomp profile defaulting, you must run the kubelet with the SeccompDefault [feature gate](https://v1-25.docs.kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/) enabled (this is the default). You must also explicitly enable the defaulting behavior for each node where you want to use this with the corresponding --seccomp-default [command line flag](https://v1-25.docs.kubernetes.io/docs/reference/command-line-tools-reference/kubelet)."
  - Seems like a user must explicitly configure some things before it all starts kicking in.

So, I'd rather not add this field by default.

Reply:
Also:
- "It is possible that the default profiles differ between container runtimes and their release versions, for example when comparing those from CRI-O and containerd."

Reply:
So I had to read the docs again and the possibility to use seccompprofile was made GA in 1.19.
https://kubernetes.io/docs/tutorials/security/seccomp/

The new feature that becomes beta in 1.25 is that we don't have to set this value, instead it will apply by default to all pods unless you specifically say no.
When this feature becomes GA we can remove this config from the operator, sadly we have to wait for GA since the enablement of beta features was disabled in 1.25 or 1.26 (can't remember which one).

Containerd, docker and cri-o all have defaultprfoiles and I assume all CNI:s do.
And we can see in the docs of OCP that they even apply the default seccompprofile after OCP 4.11 https://docs.openshift.com/container-platform/4.11/security/seccomp-profiles.html
But they use the old annotation solution, but they can be used at the same time, at least from a kubernetes point of view.

So I don't see any issues with adding seccompprofile as default the grafana instance.

Reply:
@NissesSenap Ah, I guess I misunderstood the docs then, sorry for that :)
If that's the case, then I guess we can add it back :)

Reply:
No worries at all :)